grammar.js

```diff
@@ -373,6 +378,39 @@ module.exports = grammar({
       seq(repeat($._annotation), '[', ']')
     )),
 
+    switch_construct: $ => seq(
+      'switch',
+      field('condition', $.parenthesized_expression),
+      field('body', $.switch_block)
+    ),
+
+    switch_block: $ => seq(
+      '{',
+      choice(repeat($.switch_line), repeat($.arrow_switch_line)),
+      '}'
+    ),
+
+    switch_line: $ => choice(
+      $.switch_label, 
+      $.statement
+    ),
+
+    switch_label: $ => choice(
+      seq('case', $.expression, ':'),
+      seq('default', ':')
+    ),
+
+    arrow_switch_line: $ => seq(
+      $.arrow_switch_label,
+      '->',
+      choice($.expression_statement, $.throw_statement, $.block)
+     ),
+
+    arrow_switch_label: $ => choice(
```

Comment:
In the spec, they seem to imply that this comma-separated structure can be used *both* in the arrow rules and in the regular case labels. What do you think about unifying the `switch_label` and `arrow_switch_label` rules?

Reply:
That seems reasonable, but I run into this conflict:

```
Unresolved conflict for symbol sequence:

  'switch'  parenthesized_expression  '{'  'default'  •  'class'  …

Possible interpretations:

  1:  'switch'  parenthesized_expression  '{'  (modifiers_repeat1  'default')  •  'class'  …
  2:  'switch'  parenthesized_expression  '{'  (switch_label  'default')  •  'class'  …
```

So

```

    switch_line: $ => choice(
      $.switch_label,
      ':', 
      $.statement
    ),

    switch_label: $ => choice(
      seq('case', commaSep1($.expression)),
      seq('default')
    ),

    switch_labeled_rule: $ => seq(
      $.arrow_switch_label,
      '->',
      choice($.expression_statement, $.throw_statement, $.block)
     ),

    arrow_switch_label: $ => choice(
      seq('case', commaSep1($.expression)),
      'default'
    ),
```

does not cause a conflict, but

```

    switch_line: $ => choice(
      $.switch_label, 
      ':',
      $.statement
    ),

    switch_labeled_rule: $ => seq(
      $.switch_label,
      '->',
      choice($.expression_statement, $.throw_statement, $.block)
     ),

    switch_label: $ => choice(
      seq('case', commaSep1($.expression)),
      'default'
    ),
```

does

Reply:
(Thanks for pointing out that both switches use that case label form, though, I missed that!)